cirq/protocols/apply_unitary.py

```diff
@@ -86,28 +90,60 @@ def __init__(self,
                 dtype as the target tensor.
             axes: Which axes the unitary effect is being applied to (e.g. the
                 qubits that the gate is operating on).
+            qid_shape: A tuple of the dimensions of each qid in the state
+                `target_tensor`.
         """
+        self.axes = tuple(axes)
+        if qid_shape is None:
+            # Guess what the qid_shape would be for qubits.
+            min_num_qubits = max(self.axes, default=-1) + 1
+            qid_shape = (2,) * min_num_qubits  # Might be too short
```

Comment:
Because users would be unlikely to make this particular value, we can make the argument required by the constructor. This will avoid this very bad case where the shape is wrong.